stl/inc/flat_set

```diff
@@ -228,25 +235,25 @@ public:
         }
     }
 
-    auto insert(const value_type& _Val) {
+    auto insert(const _Kty& _Val) {
         return _Emplace(_Val);
     }
-    auto insert(value_type&& _Val) {
+    auto insert(_Kty&& _Val) {
         return _Emplace(_STD move(_Val));
     }
-    template <class _Other>
+    template <_Different_from<_Kty> _Other>

```

Comment:
For example, without this constraint, when the comparer is transparent inputs like `non-const _Kty&` will go into this function. The case will be correctly handled by `_Emplace`, but I think it will be better just fall to `insert(const _Kty)`.